front/src/modules/timesStops/hooks/useOutputTableData.ts

```diff
@@ -87,44 +95,60 @@ const useOutputTableData = (
         positionOnPath: path.path_item_positions[index],
       };
     });
-  }, [selectedTrainSchedule, path, trainSummary?.pathItemTimes, startDatetime]);
-
-  const rows: TimeStopsRow[] = useMemo(() => {
-    if (!operationalPoints || !startDatetime) return [];
+  }, [selectedTrainSchedule, path, trainSummary?.pathItemTimes]);
 
-    return operationalPoints.map((op) => {
-      const matchingPathStep = pathStepRows.find(
-        (pathStepRow) => op.position === pathStepRow.positionOnPath
-      );
-      if (matchingPathStep) {
-        return {
-          ...matchingPathStep,
-          opId: op.id,
-          name: op.extensions?.identifier?.name,
-          ch: op.extensions?.sncf?.ch,
-        };
+  useEffect(() => {
+    const formatRows = async () => {
+      if (!operationalPoints || !startDatetime) {
+        setRows([]);
+        return;
       }
 
-      // compute arrival time
-      const matchingReportTrainIndex = simulatedTrain.positions.findIndex(
-        (position) => position === op.position
+      const formattedPoints = formatSuggestedOperationalPoints(operationalPoints);
```

Comment:
Ah hum the type SuggestedOperationalPoint is very messy and I wish to avoid spreading it. Can you make the function addTrackNames more generic so it takes only operationalPoints which contain at least a trackId and returns the same list of operationalPoints with their trackNames ?

It would be smth like this:
```
export const addTrackNamesToOperationalPoints = async <T extends { track: string}>(
  operationalPoints: T[],
  trackSections: InfraObjectWithGeometry[]
): Promise<(T & { trackName?: string})[]> => {...}
```